clients/rust/src/models/resource_label.rs

```diff
@@ -0,0 +1,34 @@
+/*
+ * Fermyon Cloud API
+ *
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0
+ * 
+ * Generated by: https://openapi-generator.tech
+ */
+
+
+
+
+#[derive(Clone, Debug, PartialEq, Default, Serialize, Deserialize)]
+pub struct ResourceLabel {
+    #[serde(rename = "label")]
+    pub label: String,
+    #[serde(rename = "appId")]
+    pub app_id: uuid::Uuid,
+    #[serde(rename = "appName", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
+    pub app_name: Option<Option<String>>,
```

Comment:
Why is this `Option<Option<String>>`?

Reply:
because in the service, it is set as an optional nullable field (`public string? AppName { get; set; }`) since it is not always set. Would it be better to disallow it being nullable and make it an empty string instead?

Reply:
I'm confused... Optional and nullable? Doesn't nullability hint at it being optional? I just don't get why it's not `Option<String>` instead...

Reply:
Seems like a bug. I will submit a fix upstream.

Reply:
To elaborate, this is an oddity related to how C# code is mapped to the OpenAPI schema. A property marked as `[Required]` indicates to the generator to mark it as a `required` property in the OpenAPI schema.

Because it was not marked as `[Required]` and it was a nullable string (`string?`), we got this as the OpenAPI document:

```
      "ResourceLabel": {
        "required": [
          "appId",
          "label"
        ],
        "type": "object",
        "properties": {
          "label": {
            "minLength": 1,
            "type": "string"
          },
          "appId": {
            "type": "string",
            "format": "uuid"
          },
          "appName": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
      },
```

Notice how it was not part of the `required` parameters and it's marked as `nullable: true`.

By changing it to a `string`, we get this:


```
      "ResourceLabel": {
        "required": [
          "appId",
          "label"
        ],
        "type": "object",
        "properties": {
          "label": {
            "minLength": 1,
            "type": "string"
          },
          "appId": {
            "type": "string",
            "format": "uuid"
          },
          "appName": {
            "type": "string"
          }
        },
        "additionalProperties": false
      },
```

It's a very subtle change: `appName` is still not a `required` parameter, but now it is not `nullable`.

This is why the Rust client generated it as an `<Option<Option<String>>`: the first `Option` represents whether it is a required parameter. The second option represents the nullability of the type.

Reply:
For what it's worth, it took me a while to figure this out the first time this cropped up.